src/ophys_etl/utils/array_utils.py

```diff
@@ -0,0 +1,261 @@
+                h5f['data'],
+                input_fps,
+                output_fps,
+                strategy,
+                random_seed)
+    return video_out
+
+
+def transform_to_webm(video: np.ndarray, output_path: str,
+                      fps: float, ncpu: int, bitrate: str = "0",
+                      crf: int = 20) -> str:
+    """Function to transform 2p gray scale video into a webm
+    video using imageio_ffmpeg.
+
+    Parameters
+    ----------
+    video : np.ndarray
+        Video to be transformed with shape (time, row, col)
+    output_path : str
+        Output path for the transformed video
+    fps : float
+        Desired frames per second (fps) of the output video
+    ncpu : int
+        Degree of parallelization desired for encoding. Video will be
+        split into 'ncpu' parts and each part will be encoded in parallel.
+    bitrate : str, optional
+        Desired bitrate of output, by default "0". The default *MUST*
+        be zero in order to encode in constant quality mode.
+    crf : int, optional
+        Desired perceptual quality of output, by default 20. Value can
+        be from 0 - 63. Lower values mean better quality.
+
+    Returns
+    -------
+    str
+        Output path of the encoded video
+    """
+
+    split_video = np.array_split(video, ncpu)
+    split_output_paths = [tempfile.NamedTemporaryFile(suffix=f"_{i}.webm")
+                          for i in range(ncpu)]
+
+    mp_pool_args = [(vid, outpath.name, fps, bitrate, crf)
+                    for vid, outpath in zip(split_video, split_output_paths)]
+
+    with multiprocessing.Pool(ncpu) as pool:
+        encode_results = pool.starmap(encode_video, mp_pool_args)
+
+    concat_output = concat_videos(encode_results, output_path)
+
+    for sop in split_output_paths:
+        sop.close()
+
+    return concat_output
+
+
+def concat_videos(video_paths: List[str], output_path: str) -> str:
+    """Use ffmpeg to concatenate a list of videos (with the same encoding)
+    into a single video.
+
+    Parameters
+    ----------
+    video_paths : List[str]
+        A list of paths (str) for videos that should be concatenated together
+        Order of paths matters!
+    output_path : str
+        The desired output path for the concatenated video
+
+    Returns
+    -------
+    str
+        Path of concatenated video
+    """
+
+    with tempfile.NamedTemporaryFile(suffix=".txt") as concat_list_file:
+        with open(concat_list_file.name, 'w') as fp:
+            for path in video_paths:
+                fp.write(f"file '{path}'\n")
+
+        # See: https://trac.ffmpeg.org/wiki/Concatenate
+        ffmpeg_concat_cmd = [mpg.get_ffmpeg_exe(), '-y', '-f', 'concat',
+                             '-safe', '0', '-i', concat_list_file.name,
+                             '-c', 'copy', output_path]
+        subprocess.run(ffmpeg_concat_cmd)
+
+    return output_path
+
+
+def encode_video(video: np.ndarray, output_path: str,
```

Comment:
Looks like this function and other video related ones should be removed from this file now that `ophys_etl.utils.video_utils` exists?

Reply:
ooops. removed.